src/mbgl/sprite/sprite_parser.cpp

```diff
@@ -20,9 +20,10 @@ SpriteImagePtr createSpriteImage(const PremultipliedImage& image,
                                  const double ratio,
                                  const bool sdf) {
     // Disallow invalid parameter configurations.
-    if (width == 0 || height == 0 || ratio <= 0 || ratio > 10 || width > 1024 ||
-        height > 1024) {
-        Log::Warning(Event::Sprite, "Can't create sprite with invalid metrics");
+    if (width <= 0 || height <= 0 || width > 1024 || height > 1024 ||
+        ratio <= 0 || ratio > 10 ||
+        srcX < 0 || srcY < 0 || srcX + width > image.width || srcY + height > image.height) {
```

Comment:
Can `srcX + width` or `srcY + height` overflow? Should we check that `width` and `height` are multiples of `ratio`?